### DISCUSSION TITLE
Is there a way to get fields from expression?

### DISCUSSION THREAD

**Discussion by jiachengxu** [antonmedv/expr#331] [Q&A]
For example,
the expression: `foo.boo + foo.foo.boo`
get fields which are mentioned in the expression: `["foo.boo", "foo.foo.boo" ]`

**Comment by antonmedv** [comment]
Possible. Use https://github.com/antonmedv/expr/blob/master/docs/Visitor-and-Patch.md

**Reply by jiachengxu** [reply]
Thank you!
